### PR TITLE
WebSocket server API fixes

### PR DESCRIPTION
**Pull request by sonnyp**

### CODE REVIEW

File: lib/websocket/server.js [hunk 1]
```diff
@@ -25,13 +25,21 @@ util.inherits(WsServer, CS2Server)
 
 WsServer.prototype.C2SStream = C2SStream
 
-WsServer.prototype.start = function() {
+WsServer.prototype.listen = function() {
   var self = this
```
Comment (chris-rock) [inline lib/websocket/server.js]: can we return wss instance for listen() ?

Comment (sonnyp) [inline lib/websocket/server.js]: That's a good idea but AFAIK we don't for others transport (TCP and BOSH) maybe a new PR that returns the underlying server for all transports?

File: lib/websocket/server.js [hunk 2]
```diff
@@ -25,13 +25,21 @@ util.inherits(WsServer, CS2Server)
...
   this.wss = new WebSocketServer({
     port: self.wsoptions.port
   })
 
+  this.wss.once('listening', function() {
+    self.emit('online')
+  })
+
+  this.wss.once('error', function(err) {
+    self.emit('error')
```
Comment (sonnyp) [inline lib/websocket/server.js]: oops, I should pass the err as well, we'll edit

File: lib/websocket/server.js (hunk 1 above)
Comment (chris-rock) [inline lib/websocket/server.js]: sounds great

**Comment by sonnyp** [comment]
@chris-rock fixed the err thing

File: lib/websocket/server.js [hunk 3]
```diff
@@ -25,13 +25,21 @@ util.inherits(WsServer, CS2Server)
...
   this.wss = new WebSocketServer({
     port: self.wsoptions.port
   })
 
+  this.wss.once('listening', function() {
+    self.emit('online')
+  })
+
+  this.wss.once('error', function(err) {
+    self.emit('error', err)
```
Comment (lloydwatkin) [inline lib/websocket/server.js]: err === error ;)

**Comment by sonnyp** [comment]
@lloydwatkin done

**Comment by chris-rock** [comment]
cool. Thanks @sonnyp

**Pull request merged**